### ISSUE TITLE
Add a “Polls only” option to local and federated timelines

### ISSUE THREAD

**Issue by nclm** [tootsuite/mastodon#10436]
### Pitch

The local and federated timelines have a “Media only” option to only look at images and videos. Now that polls exists as a new kind of post, there could be an additional “Polls only”.

![Polls only](https://user-images.githubusercontent.com/923008/55292344-dbb3f480-53e9-11e9-85a3-0e8b4f397c3d.png)

### Motivation

Could be nice for users that are in the mood for answering to polls :) (a little like being in the mood for filling-up surveys and going to /r/samplesize, for instance).

**Comment by Cassolotl** [comment]
I love polls so I agree, but I suspect that might make the timeline *very* quiet...!

**Comment by StarryAster** [comment]
This seems to me like a good option that would give more visibility to polls (for people who don't/can't keep track of all of the posts on the local and/or federated timelines but who like answering polls).

And since anyone would have to click on that option, I don't think that would be something that could get many people confused as to why are their local/federated timelines so empty.

**Comment by nightpool** [comment]
it's very easy to get confused—i once spent 4 months wondering why the
cybre space local timeline was so empty before realizing that I had clicked
"media only" at some point in the past and forgotten about it.

**Comment by trwnh** [comment]
might be a good idea to make filters have a UI similar to the notification quick-filter, but clicking should toggle enable-disable of certain types of posts. this could/should override the "show x" toggles in the column settings, if the quick-filter is visible.

**Comment by nclm** [comment]
Yes, this UI option is far from optimal, maybe I should have titled the issue “Add a way to browse public currently-running polls”, since adding a “Polls only“ switch there is only one solution.

It all depends of how much we consider a “looking at polls” option (and a “looking at media” one, for that matter) central or not, how much we want it discoverable. 

I think it is a scale like this:

1. **Not important.** No option to browse polls.

2. **Niche.** Possible solution: just add a “Polls only” option in the submenu of public timelines, alongside “Media only”. People who know it’s here or are curious enough to click everywhere might use it, but it is not really discoverable. Downside is indeed that it can stay checked and hidden causing the timelines to seem broken.

![Screen Shot 2019-03-31 at 19 14 53](https://user-images.githubusercontent.com/923008/55422059-100aea80-557b-11e9-97a7-121de35dd74c.png)

3. **Nice.** Possible solution: add a tab bar similar to the one on profiles to the public timelines columns to select in between All, Media only, and Polls only. Far more discoverable since the options are always visible.

![Screen Shot 2019-04-02 at 19 04 24](https://user-images.githubusercontent.com/923008/55422320-accd8800-557b-11e9-9a00-ff8c607f4163.png)

4. **Important.** Polls are an essential feature. Possible solution: add a dedicated column for the polls. It would list all the currently running polls (the ones your instance has access to), show you the results of the polls you took part in, etc.

![Screen Shot 2019-04-02 at 19 15 19](https://user-images.githubusercontent.com/923008/55422418-dbe3f980-557b-11e9-816e-2f016fda541a.png)

For me, 1 (current) might not be enough, 4 is fun but too much (though it could be an admin-activated feature on a per-instance basis?). Ideally, I’d love a 2.5, better than 2 but not as visible as 3.

**Comment by ealgase** [comment]
I think 3 would be pretty nice and the tiny bar of space it would take would be well worth the more convenient functionality.
